docs/utilities/batch.md

```diff
@@ -473,4 +473,95 @@ see the details for [SQS](https://docs.aws.amazon.com/lambda/latest/dg/with-sqs.
             }
           ]
         }
-    ```
\ No newline at end of file
+    ```
+
+
+## Handling Messages
+
+### Raw message and deserialized message handlers
+You must provide either a raw message handler, or a deserialized message handler. The raw message handler receives
+the envelope record type relevant for the particular event source - for instance, the SQS event source provides
+[SQSMessage](https://javadoc.io/doc/com.amazonaws/aws-lambda-java-events/2.2.2/com/amazonaws/services/lambda/runtime/events/SQSEvent.html)
+instances. The deserialized message handler extracts the body from this envelope, and deserializes it to a user-defined
+type. 
+
+In general, the deserialized message handler should be used unless you need access to information on the envelope.
+
+=== "Raw Message Handler"
+
+    ```java
+    public void setup() {
+        BatchMessageHandler<SQSEvent, SQSBatchResponse> handler = new BatchMessageHandlerBuilder()
+                .withSqsBatchHandler()
+                .buildWithRawMessageHandler(this::processRawMessage);
+    }
+
+    private void processRawMessage(SQSEvent.SQSMessage sqsMessage) {
+        // Do something with the raw message
+    }
+    
+    ```
+
+=== "Deserialized Message Handler" 
+
+    ```java
+    public void setup() {
+        BatchMessageHandler<SQSEvent, SQSBatchResponse> handler = new BatchMessageHandlerBuilder()
+                .withSqsBatchHandler()
+                .buildWitMessageHandler(this::processRawMessage, Product.class);
+    }
+
+    private void processMessage(Product product) {
+        // Do something with the raw message
+    }
+    
+    ```
+
+### Success and failure handlers
+
+You can register a success or failure handler which will be invoked as each message is processed by the batch
+module.
+
+Handlers can be provided when building the batch processor and are available for all event sources.
+For instance for DynamoDB:
+
+```java
+BatchMessageHandler<DynamodbEvent, StreamsEventResponse> handler = new BatchMessageHandlerBuilder()
+            .withDynamoDbBatchHandler()
+            .withSuccessHandler((m) -> {
+                // Success handler receives the raw message
+                LOGGER.info("Message with sequenceNumber {} was successfully processed",
+                    m.getDynamodb().getSequenceNumber());
+            })
+            .withFailureHandler((m, e) -> {
```

Comment:
maybe add a note that they don't have to handle the partialBatchFailure as it is already handled by the library